### ISSUE TITLE
Check the optimization strategies

### ISSUE THREAD

**Issue by tobias-liaudat** [CosmoStat/wf-psf#5]
We need to choose an appropriate optimization strategy to train our model once issue #2 is finished.

This includes specifying:
- Optimiser
- Batch size
- Number of epochs

Check performance on the recovery of the PSF field polynomial coefficients_matrix.

At first we should continue to do this with the noiseless correctly sampled PSFs. Once we observe that the optimization strategy is working we can degrade the observations. First noise, then downsampling. To see.

**Comment by tobias-liaudat** [comment]
It seems the parametric model is working!

- The batch size is limited by the GPU memory specs.
  -  Using float32 and a OPD dim size of 256 I could use a batch size of 16 without any problems (on the different GPU's google colab assigned me, 10Gb, 14Gb, 15Gb).

Right now, trying with a standard ML optimiser like Adam (with default parameters, learning rate of 1e-2). Seems to be working fine.

For now I'm only doing qualitative comparisons of the results. Later, I have to implement quantitative comparisons.

I got the following results.

### Training

<img width="825" alt="Screenshot 2021-01-26 at 14 33 05" src="https://user-images.githubusercontent.com/16106854/105852335-9f04b480-5fe4-11eb-8cb2-cf8cca56a35d.png">

### Reconstruction of one PSF

<img width="670" alt="Screenshot 2021-01-26 at 14 32 52" src="https://user-images.githubusercontent.com/16106854/105852388-b0e65780-5fe4-11eb-95a1-632f6d5dbcec.png">

### Model's matrix coefficient estimations

<img width="1096" alt="Screenshot 2021-01-26 at 14 32 41" src="https://user-images.githubusercontent.com/16106854/105852428-c0fe3700-5fe4-11eb-9265-be0c35f2f691.png">
